envoy/tcp/async_tcp_client.h

```diff
@@ -50,6 +50,12 @@ class AsyncTcpClient {
    */
   virtual void close() PURE;
 
+  /**
+   * Close the client. It closes the connection based on close type.
+   * @param type the connection close type.
+   */
+  virtual void close(Network::ConnectionCloseType type) PURE;
```

Comment:
Given that this is new API and there is no need for backward compatibility, I would just go with the one `close` method that requires the close type parameter.